From d285aa394dd32a69f7bf7366aab34665944b07e7 Mon Sep 17 00:00:00 2001
From: Johan Mabille <johan.mabille@gmail.com>
Date: Mon, 14 Jun 2021 10:52:42 +0200
Subject: [PATCH] Fixed number version on other UNIX-like platforms

---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 903dcf9..7da11e0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -267,10 +267,12 @@ macro(xeus_create_target target_name linkage output_name)
         endif ()
     endif ()
 
-    if (UNIX AND NOT APPLE)
+    if (UNIX)
         # CMake does not compute the version number of so files as libtool
-        # does on Linux. Strictly speaking, we should exclude FreeBSD from
-        # this conditional branch.
+        # does on Linux. Strictly speaking, we should exclude FreeBSD and
+        # Apple from this, but that would require having different version
+        # numbers depending on the platform. We prefer to follow the
+        # libtool pattern everywhere.
         math(EXPR XEUS_BINARY_COMPATIBLE "${XEUS_BINARY_CURRENT} - ${XEUS_BINARY_AGE}")
         set_target_properties(
             ${target_name}
-- 
2.26.2

